python/flatbuffers/builder.py

```diff
@@ -510,13 +510,21 @@ def Slot(self, slotnum):
         self.current_vtable[slotnum] = self.Offset()
     ## @endcond
 
-    def __Finish(self, rootTable, sizePrefix):
+    def __Finish(self, rootTable, sizePrefix, file_identifier=None):
         """Finish finalizes a buffer, pointing to the given `rootTable`."""
         N.enforce_number(rootTable, N.UOffsetTFlags)
-        prepSize = N.UOffsetTFlags.bytewidth
-        if sizePrefix:
-            prepSize += N.Int32Flags.bytewidth
-        self.Prep(self.minalign, prepSize)
```

Comment:
I was just noticing the removal of this code, it means that possibly the buffer as a whole is misaligned?
If minalign==8, and we're writing between 1 to 3 32-bit entities (file identifier, root offset, size), those 3 need to be pre-padded to make this alignment happen.
@rw @pyottamus

Reply:
Fix would be to add this back in?
```
prepSize = N.UOffsetTFlags.bytewidth
if file_identifier is not None:
    prepSize += N.Int32Flags.bytewidth
if sizePrefix:
    prepSize += N.Int32Flags.bytewidth
self.Prep(self.minalign, prepSize)
```

Reply:
@pyottamus WDYT?

Reply:
@aardappel, @rw I removed that because both `PrependInt32` and `PrependUOffsetTRelative` handle alignment. Since the `UOffset` is called first, and `UOffset` is at least 32bits(may become 64bit in the future), the PrependInt32 will be aligned already.

Reply:
@aardappel, @rw That being said, I did notice that the part where I add the `file_identifier` might break if a 64bit `UOffset` is used. Im looking into that, and if It is a problem, I'll open up an issue and do a pull request for a fix.

Reply:
@pyottamus those calls align to their own size, what is required here is pre-alignment to `minalign`, which may be bigger.

Reply:
And none of these values are going to become 64-bit any time soon (without a format fork).